src/Compilers/Core/Portable/Syntax/SyntaxTokenList.cs

```diff
@@ -357,7 +363,8 @@ public SyntaxTokenList InsertRange(int index, IEnumerable<SyntaxToken> tokens)
                 return this;
             }
 
-            return new SyntaxTokenList(null, list[0].Node.CreateList(list.Select(n => n.Node)), 0, 0);
+            Debug.Assert(list[0].Node is object);
+            return new SyntaxTokenList(null, list[0].Node!.CreateList(list.Select(n => n.Node!)), 0, 0);
```

Comment:
I'm not sure how we justify this suppression inside `Select`

Reply:
@jcouv sorry I meant to leave a comment here for us to discuss. Basically I'm not sure what the correct way to handle this situation is. The `n.Node` values can come from two sources here:

1. Nodes already in the list. These are non-null by virtue of going through `CreateList` previously which provides this guarantee. 
1. Nodes being inserted into the list. These have no guarantees and no obvious place to put them here. 

I'm not sure what the best pattern is for annotating this contract. A loop seemed excessive here. I thought about changing to use a `RequiredNode` pattern. That seems more appropriate but wasn't sure if it was overkill or not here. It's a knarly part of our nullable puzzle. 

Very open to  suggestions.

Reply:
Using `RequiredNode` as discussed offline